requirements/specifications/lifecycle/index.md

```diff
@@ -479,81 +682,255 @@ scope for this document.
+> The platform **MUST** dispatch the `Lifecycle.onRequestDestroy`
+> notification to the app, and wait for `appDestroyTimeout` milliseconds
+> for either a `Lifecycle.destroyResult` or `Lifecycle.destroyError`
+> call in response.
+>
+> Once the platform receives the `destroyResult` call, then the platform
+> may proceed with app unloading.
+>
+> If the app times out or makes an `destroyError` call, then the app
+> **MUST** be terminated.
 
-### 3.8. Terminating an app
+### 4.10. Terminating an app
 
 Terminating an app removes it from memory without dispatching any state
 changes.
 
-The platform may terminate apps for any number of reasons that are out
-of scope for this document, e.g. an app becomes non-responsive.
+The platform **SHOULD** terminate apps when one of the transitions outlined
+here fails for some reason, or if an app becomes non-responsesive.
 
 Terminating an app results in removing it from memory and all resources
 held by the app and it's container being freed up.
 
 Platforms **MAY** terminate an app when needed but **SHOULD NOT** do
-this in place of graceful unloading.
-
-## 4. Core SDK APIs
+this in place of graceful [unloading](#38-unloading-an-app).
+
+## 5. Getting the current status
+The Lifecycle module **MUST** provide a `status` property API that
+returns the current lifecycle transition of the app if one is in
+progress, otherwise, it returns the current state.
+
+If an app is in a transtition from one state to another, then it
+**MUST** be considered in the transition until
+such time as the app's implementation of the [transition](#4-lifecycle-state-transitions) has returned
+and the platfrom has finshed the transition.
+
+The `status` API **MUST** have a corresponding `onStatusChanged`
+notification.
+
+The `status` API must have one of the following values:
+
+- `Initializing`
+- `Creating`
+- `Running`
+- `Suspending`
+- `Suspended`
+- `Resuming`
+- `Activating`
+- `Active`
+- `Deactivating`
+- `Sleeping`
+- `Waking`
+- `Destroying`
+
+## 6. Activity Navigation
+Typically navigation is handled either when the app is activated, via
+the `intent` parameter of the `activate` method, or by internal input
+within the app.
+
+There are other times when the platform needs to inform an app of a user's
+intent to navigate when the app is already `ACTIVE`, e.g. when a voice
+command is executed or a soft remote sends a message while the app is in
+focus.
+
+In these cases, the platform **MUST** call the `Activity.navigate` method
+of the App, and pass the `intent`.
+
+Within the `navigate()` method, the app **MUST** inspect the `intent`
+parameter and prepare to fulfill a specific [Navigation Intent](../intents/navigation.md) for
+the app, which may include:
+
+ - Loading any metadata needed to display the user's intended content.
+ - Performing any entitlement checks to decide whether to display a player or a purchase flow
+ - Any other steps necesary to present content to the user quickly
+
+## 7. Launch Parameters
```

Comment:
I think we should avoid the term "Launch" since it is ambiguous with the states.
Maybe this should be called `Create Parameters`?

Reply:
Discuss and agreed to keep  as is